sdk/search/search-documents/test/public/utils/recordedClient.ts

```diff
@@ -53,6 +53,15 @@ export const environmentSetup: RecorderEnvironmentSetup = {
 };
 
 export function createClients<IndexModel>(indexName: string): Clients<IndexModel> {
+  switch (testEnv.AZURE_AUTHORITY_HOST) {
+    case "https://login.microsoftonline.us":
+      process.env.ENDPOINT = process.env.ENDPOINT!.toString().replace(".windows.net", ".azure.us");
+      break;
+    case "https://login.chinacloudapi.cn":
+      process.env.ENDPOINT = process.env.ENDPOINT!.toString().replace(".windows.net", ".azure.cn");
+      break;
```

Comment:
I am not very sure about this solution. The tests should not be in the business of changing endpoints. If the input to the tests is going to use a different authority host in order to target other clouds, then the endpoint that gets passed in should be different too.

In #17424, we discuss about updating the `cognitiveServicesEndpointSuffix` such that the output from the template is using the right endpoint. Can we not do the same for search?

cc @benbp

Reply:
@ramya-rao-a I know I just linked to this, but I agree for test code it should contain as little knowledge as possible about endpoints. They should either be set by the credential builder code, or by the test environment.

Reply:
@ramya-rao-a Do you want me to revert, or will you take this as a follow-on?

Reply:
If you can log an issue for search, then @sarangan12 can follow up on that

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/17577